spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -192,6 +194,14 @@ public Map<String, String> extraSnapshotMetadata() {
           }
         });
 
+    final Iterator<Tuple2<String, String>> it = sessionConf.getAll().iterator();
+    while (it.hasNext()) {
```

Comment:
Rather than using Scala objects, which are awkward in Java, we like to use Scala's conversions to get Java objects:

```java
import scala.collection.JavaConverters;
Map<String, String> javaMap = JavaConverters.mapAsJavaMapConverter(sessionConf.getAll()).asJava();
```

Reply:
thanks for tips @rdblue, i use scala class because i did not know mapAsJavaMapConverter :)